homeassistant/components/media_player/__init__.py

```diff
@@ -674,6 +675,31 @@ def state_attributes(self):
 
         return state_attr
 
+    def preload_media_image_url(self, image_url):
+        """Preload and cache a media image for future use."""
+        self.hass.loop.call_later(
+            0.1,
+            _fetch_image,
+            image_url
+        )
+
+
+@functools.lru_cache(maxsize=16)
+def _fetch_image(url):
+    """Helper method to fetch image.
+
+    Up to 16 images are cached in memory (the images are typically 10-100kB in
+    size).
+    """
+    try:
+        response = requests.get(url)
```

Comment:
@pvizeli I don't think `lru_cache` works with coroutines. We may need to write our own, something like; http://stackoverflow.com/a/34123396. We should also consider using `wrapt` (https://hynek.me/articles/decorators/)